### PR TITLE
CIRC-1078 Block request when expiration date is not set

### PR DESCRIPTION
**Pull request by andrew-khodyka**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 1]
```diff
@@ -1655,7 +1655,34 @@ public void cannotCreateRequestWhenRequesterHasActiveRequestManualBlocks() {
     assertThat(postResponse.getJson(), hasErrors(1));
     assertThat(postResponse.getJson(), hasErrorWith(allOf(
       hasMessage("Patron blocked from requesting"))));
-      hasParameter("reason", "Display description");
+    hasParameter("reason", "Display description");
+  }
+
+  @Test
+  public void cannotCreateRequestWhenExpirationDateIsNotSet() {
```
Comment (marcjohnson-kint) [inline src/test/java/api/requests/RequestsAPICreationTests.java:1662]: Does this test demonstrate that any request cannot be created without an expiration date or only requests for a patron that has been manually blocked from creating requests?

File: src/main/java/org/folio/circulation/domain/validation/UserManualBlocksValidator.java [hunk 2]
```diff
@@ -60,6 +60,6 @@ private boolean isUserBlockedManually(MultipleRecords<UserManualBlock> userManua
 
   private boolean isBlockedToCreateRequests(DateTime expirationDate, boolean requests) {
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/validation/UserManualBlocksValidator.java:61]:
```suggestion
  private boolean isBlockedToCreateRequests(DateTime expirationDate, boolean blockRequests) {
```

File: src/main/java/org/folio/circulation/domain/validation/UserManualBlocksValidator.java [hunk 3]
```diff
@@ -60,6 +60,6 @@ private boolean isUserBlockedManually(MultipleRecords<UserManualBlock> userManua
...
     final DateTime now = ClockManager.getClockManager().getDateTime();
-    return requests && expirationDate != null && expirationDate.isAfter(now);
+    return expirationDate == null || requests && expirationDate.isAfter(now);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/validation/UserManualBlocksValidator.java:63]:
```suggestion
    return requests && (expirationDate == null || expirationDate.isAfter(now));
```

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 4]
```diff
@@ -1655,7 +1655,34 @@ public void cannotCreateRequestWhenRequesterHasActiveRequestManualBlocks() {
...
+    checkOutFixture.checkOutByBarcode(item, usersFixture.jessica());
+    userManualBlocksFixture.create(userManualBlockBuilder);
+
+    Response postResponse = requestsClient.attemptCreate(requestBuilder);
+
+    assertThat(postResponse, hasStatus(HTTP_UNPROCESSABLE_ENTITY));
+    assertThat(postResponse.getJson(), hasErrors(1));
+    assertThat(postResponse.getJson(), hasErrorWith(allOf(
+      hasMessage("Patron blocked from requesting"))));
+    hasParameter("reason", "Display description");
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/requests/RequestsAPICreationTests.java:1685]: Please fix this dangling matcher

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 5]
```diff
@@ -1654,8 +1654,37 @@ public void cannotCreateRequestWhenRequesterHasActiveRequestManualBlocks() {
...
+    final UUID pickupServicePointId = servicePointsFixture.cd1().getId();
+
+    final UserManualBlockBuilder userManualBlockBuilder = getManualBlockBuilder()
+      .withRequests(true)
+      .withUserId(String.valueOf(requester.getId()));
+
+    final RequestBuilder requestBuilder = createRequestBuilder(item,
+      requester,
+      pickupServicePointId,
+      new DateTime(2017, 7, 22, 10, 22, 54, DateTimeZone.UTC));
```
Comment (alexanderkurash) [inline src/test/java/api/requests/RequestsAPICreationTests.java:1675]: We can create a PAGE request and avoid check-out action. Also, we can move this part directly before `requestsClient.attemptCreate`. Example:

``` 
new RequestBuilder()
  .withId(UUID.randomUUID())
  .open()
  .page()
  .forItem(item)
  .by(requester)
  .fulfilToHoldShelf()
  .withPickupServicePointId(pickupServicePointId);
```

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 6]
```diff
@@ -1654,8 +1654,37 @@ public void cannotCreateRequestWhenRequesterHasActiveRequestManualBlocks() {
...
+
+  @Test
+  public void cannotCreateRequestWhenRequesterHasActiveRequestManualBlocksAndExpirationDateIsNotSet() {
+    final IndividualResource item = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource requester = usersFixture.rebecca();
+    final UUID pickupServicePointId = servicePointsFixture.cd1().getId();
+
+    final UserManualBlockBuilder userManualBlockBuilder = getManualBlockBuilder()
+      .withRequests(true)
+      .withUserId(String.valueOf(requester.getId()));
```
Comment (mtraneis) [inline src/test/java/api/requests/RequestsAPICreationTests.java:1670]: Can a comment be added here to highlight that the manual block's expiration date is intentionally not been set?

**Pull request closed without merging**